src/System.Text.Primitives/System/Text/Encoders/UnicodeScalar.cs

```diff
@@ -0,0 +1,246 @@
+        /// <summary>
+        /// Converts an <see cref="int"/> to a <see cref="UnicodeScalar"/>.
+        /// Identical to calling <see cref="UnicodeScalar.UnicodeScalar(int)"/>.
+        /// </summary>
+        public static explicit operator UnicodeScalar(int value) => new UnicodeScalar(value);
+
+        /// <summary>
+        /// If this scalar is representable as a single UTF-16 code unit (see <see cref="Utf16CodeUnitCount"/>),
+        /// returns the scalar value as the UTF-16 code unit. Otherwise throws <see cref="OverflowException"/>.
+        /// </summary>
+        public static explicit operator char(UnicodeScalar value) => checked((char)value.Value);
+
+        /// <summary>
+        /// Returns the scalar value as a 24-bit integer. Equivalent to reading the <see cref="Value"/> field.
+        /// </summary>
+        public static explicit operator int(UnicodeScalar value) => value.Value;
+
+        public static bool operator ==(UnicodeScalar a, UnicodeScalar b) => (a.Value == b.Value);
+
+        public static bool operator !=(UnicodeScalar a, UnicodeScalar b) => (a.Value != b.Value);
+
+        public static bool operator <(UnicodeScalar a, UnicodeScalar b) => (a.Value < b.Value);
+
+        public static bool operator <=(UnicodeScalar a, UnicodeScalar b) => (a.Value <= b.Value);
+
+        public static bool operator >(UnicodeScalar a, UnicodeScalar b) => (a.Value > b.Value);
+
+        public static bool operator >=(UnicodeScalar a, UnicodeScalar b) => (a.Value >= b.Value);
+
+        private bool IsValid => IsValidScalar((uint)Value);
+
+        /// <summary>
+        /// Returns the number of UTF-16 code units (<see cref="char"/>s) required to represent this scalar.
+        /// Scalars in the range [U+0000..U+FFFF] require one UTF-16 code unit.
+        /// Scalars in the range [U+10000..U+10FFFF] require two UTF-16 code units.
+        /// </summary>
+        public int Utf16CodeUnitCount
+        {
+            get
+            {
+                Debug.Assert(IsValid);
+                return ((uint)Value < 0x10000U) ? 1 : 2;
+            }
+        }
+
+        /// <summary>
+        /// Returns the number of UTF-8 code units (<see cref="byte"/>s) required to represent this scalar.
+        /// Scalars in the range [U+0000..U+007F] require one UTF-8 code unit.
+        /// Scalars in the range [U+0080..U+07FF] require two UTF-8 code units.
+        /// Scalars in the range [U+0800..U+FFFF] require three UTF-8 code unit.
+        /// Scalars in the range [U+10000..U+10FFFF] require four UTF-8 code units.
+        /// </summary>
+        public int Utf8CodeUnitCount
+        {
+            get
+            {
+                Debug.Assert(IsValid);
+
+                if ((uint)Value < 0x80U)
+                {
+                    return 1;
+                }
+                else if ((uint)Value < 0x800U)
+                {
+                    return 2;
+                }
+                else if ((uint)Value < 0x10000U)
```

Comment:
Save 0x10000U (and others) as constants or use https://github.com/dotnet/corefxlab/blob/master/src/System.Text.Primitives/System/Text/Encoders/EncodingHelper.cs#L13?

This is a duplicate of: https://github.com/dotnet/corefxlab/blob/master/src/System.Text.Primitives/System/Text/Encoders/EncodingHelper.cs#L107
What should we do about these methods?